kinto_http/__init__.py

```diff
@@ -259,6 +267,89 @@ def delete_buckets(self, safe=True, if_match=None, if_exists=False):
+                                              group=group,
+                                              bucket=bucket,
+                                              data=data,
+                                              permissions=permissions,
+                                              safe=safe)
+        headers = DO_NOT_OVERWRITE if safe else None
+        endpoint = self.get_endpoint('group',
+                                     bucket=bucket,
+                                     group=group)
+        try:
+            resp, _ = self.session.request('put', endpoint, data=data,
+                                           permissions=permissions,
+                                           headers=headers)
+        except KintoException as e:
+            if e.response.status_code == 403:
+                msg = ("Unauthorized. Please check that the bucket exists and "
+                       "that you have the permission to create or write on "
+                       "this group.")
+                e = KintoException(msg, e)
+            raise e
+
+        return resp
+
+    def update_group(self, data=None, group=None, bucket=None,
+                     permissions=None, method='put',
+                     safe=True, if_match=None):
+        endpoint = self.get_endpoint('group',
+                                     bucket=bucket,
+                                     group=group)
+        headers = self._get_cache_headers(safe, data, if_match)
+        resp, _ = self.session.request(method, endpoint, data=data,
+                                       permissions=permissions,
+                                       headers=headers)
+        return resp
+
+    def patch_group(self, *args, **kwargs):
+        kwargs['method'] = 'patch'
+        return self.update_group(*args, **kwargs)
+
+    def get_group(self, group=None, bucket=None):
+        endpoint = self.get_endpoint('group',
+                                     bucket=bucket,
+                                     group=group)
+        resp, _ = self.session.request('get', endpoint)
+        return resp
+
+    def delete_group(self, group=None, bucket=None,
+                     safe=True, if_match=None, if_exists=False):
+        if if_exists:
+            return self._delete_if_exists('group',
+                                          group=group,
+                                          bucket=bucket,
+                                          safe=safe,
+                                          if_match=if_match)
+        endpoint = self.get_endpoint('group',
+                                     bucket=bucket,
+                                     group=group)
+        headers = self._get_cache_headers(safe, if_match=if_match)
+        resp, _ = self.session.request('delete', endpoint, headers=headers)
+        return resp['data']
+
+    def delete_groups(self, bucket=None, safe=True, if_match=None, if_exists=False):
+        if if_exists:
```

Comment:
That's a good question :) 
In short: we decided «arbitrarily» that the client should mimic the behaviour of HTTP API.

When we do a `DELETE` HTTP request on a single group that does not exist, the server returns a 404 (which is interpreted as an error).  How when we do a `DELETE` on a plural endpoint (like here `/buckets/x/groups`), the server returns an empty list if no record were deleted...

I don't say that is the only behaviour we could come up with on the server. But that's the way the server behaves currently ;)